### ISSUE TITLE
TypeError: getAndroidModule().isReadyAndroid is not a function.

### ISSUE THREAD

**Issue by chvik** [dooboolab/react-native-iap#1636]
isReadyAndroid is not implemented in the native layer. Perhaps it was left out during the migration to kotlin?

### Version of react-native-iap

8.0.1

### Version of react-native

0.64.2

### Platforms you faced the error (IOS or Android or both?)

Android

### Expected behavior

isReadyAndroid is implemented and returns a Promise<boolean>

### Actual behavior

isReadyAndroid raises a TypeError (`TypeError: getAndroidModule().isReadyAndroid is not a function.`)

### Tested environment (Emulator? Real Device?)

Nokia 5.4

### Steps to reproduce the behavior

just call `RNIap.isReadyAndroid()`

**Comment by andresesfm** [comment]
That method has been removed. All you have to do is wait for the `init` method to resolve

**Comment by chvik** [comment]
Could it be completely removed in this case?

Declared in the library API:
https://github.com/dooboolab/react-native-iap/blob/0a255579a75e64a938ecf06d6e1be3bcdbb3fdee/src/iap.ts#L678

Mentioned in the doc:
https://github.com/dooboolab/react-native-iap/blob/0a255579a75e64a938ecf06d6e1be3bcdbb3fdee/docs/docs/usage_instructions/connection_lifecycle.md?plain=1#L93
https://github.com/dooboolab/react-native-iap/blob/0a255579a75e64a938ecf06d6e1be3bcdbb3fdee/docs/docs/api_reference/methods.md?plain=1#L30

**Comment by Dallas62** [comment]
Hi @andresesfm 
Agreed with @chvik, the current API is not clear.
Is it an existing method ?
- Yes: it's not working
- No: The documentation and documentions mentioned it

The method was created to avoid an issue on Android: https://github.com/dooboolab/react-native-iap/pull/1563
Does `init` solve this issue as you mention it ?

regards

**Comment by goguda** [comment]
> Hi @andresesfm Agreed with @chvik, the current API is not clear. Is it an existing method ?
> 
>     * Yes: it's not working
> 
>     * No: The documentation mentioned it
> 
> 
> The method was created to avoid an issue on Android: #1563 Does `init` solve this issue as you mention it ?
> 
> regards

Am also wondering this, if this is the case the docs need to be updated: https://react-native-iap.dooboolab.com/docs/usage_instructions/connection_lifecycle/#long-lived-connections-android

I am getting the same error with this method when trying to use it as the docs describe.

**Comment by ifsnow** [comment]
@andresesfm Are you sure you don't need #1563 PR?

@hyochan Do you remember why #1563 was merged? If nothing else has been done about this issue in 8.0, isn't it still necessary?

@ospfranco As a PR(#1563) owner, what is your opinion?

**Comment by ospfranco** [comment]
IMO removing is bad. 

For us, it was **necessary** because we kept the android connection alive outside of a single container (we didn't just have one screen that used the object, but queried the API at different points to show different prices and special offer screens). Therefore using hooks that automatically re-initialize the connection when the component mounts/unmounts was simply not sufficient.

Since there was no way to check if the connection was alive or not we would have to re-initialize the connection every time we would need to query it. The connection got killed at "random" times (device would go to sleep, app would go to background) but not always depending if the user would re-open the app in a certain interval.

So basically... the method was **needed**, not just a convenience. But I haven't checked the codebase, if you can guarantee the google api connection is alive 100% of the time, then there is no need to expose this to JS, but if the connection can still silently die and any init method needs to be called, then there needs to be something to check the state.

**Comment by mcjambi** [comment]
i have the same issue here, isReadyAndroid is not exist, and i do not know why document has it
